.github/workflows/v2.yml

```diff
@@ -93,10 +95,77 @@ jobs:
       - name: Check that fakes are updated
         run: git diff --no-ext-diff --exit-code
 
+  ci-build-gitops-image:
+    name: V2 CI Build Gitops Docker Image
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        docker-image:
+          - gitops
+          - gitops-server
+    steps:
+      - uses: actions/checkout@v2
+      - uses: docker/setup-buildx-action@v1
+      - run: make dependencies
+      - name: Set build-time flags
+        run: |
+          echo "LDFLAGS=$(make echo-ldflags)" >> $GITHUB_ENV
+          echo "FLUX_VERSION=$(make echo-flux-version)" >> $GITHUB_ENV
+      - name: Build and export
+        uses: docker/build-push-action@v2
+        with:
+          tags: "${{ env.CI_CONTAINER_REPOSITORY }}/${{ matrix.docker-image }}:${{ github.sha }}"
+          outputs: type=docker,dest=/tmp/${{ matrix.docker-image }}.tar
+          file: ${{ matrix.docker-image }}.dockerfile
+          # Warning: the ' in build-args are fragile -- only use on LDFLAGS
+          build-args: |
+            FLUX_VERSION=${{ env.FLUX_VERSION }}
+            LDFLAGS='${{ env.LDFLAGS }}'
+            GIT_COMMIT=${{ github.sha }}
+      - name: Cache docker image
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ matrix.docker-image }}
+          path: /tmp/${{ matrix.docker-image }}.tar
+
+  ci-upload-images:
```

Comment:
Can we have this run *only* on push to the `v2` branch. I don't think we need/want to upload containers on every pull request?

Reply:
ah good point, I'll add a condition for the push step